frontend_v2/src/app/components/challenge/challengesubmissions/challengesubmissions.component.html

```diff
@@ -166,7 +166,13 @@ <h5 class="fw-light">My Submissions</h5>
                 <div class="element">
                   <mat-checkbox
                     [checked]="element.is_public"
-                    [disabled]="element.status !== 'finished' || !selectedPhase['leaderboard_public']"
+                    [disabled]="
+                      !(
+                        element.status === 'finished' ||
```

Comment:
@burnerlee the `Show on Leaderboard` option should be disabled only when submission status is either `cancelled` or `failed`